zerver/views/realm.py

```diff
@@ -54,4 +59,10 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
         do_set_realm_message_editing(realm, allow_message_editing, message_content_edit_limit_seconds)
         data['allow_message_editing'] = allow_message_editing
         data['message_content_edit_limit_seconds'] = message_content_edit_limit_seconds
+    if default_language is not None and realm.default_language != default_language:
+        if default_language in get_available_language_codes():
+            do_set_realm_default_language(realm, default_language)
+            data['default_language'] = default_language
+        else:
+            raise JsonableError(_("Invalid language '%s'" % (default_language,)))
```

Comment:
Cool, a few thoughts on this:
* From a code flow perspective, I would usually write this check as 
```
if default_language not in get_available_language_codes():
    raise JsonableError(_("Invalid language '%s'" % (default_language,)))
do_set_realm_default_language(realm, default_language)
data['default_language'] = default_language
```
since it results in less if statement depth for most of the code.
* Since this function can be requested to make multiple updates, you usually want it to either change everything or nothing.  As a result, we want the `JsonableError` check which can cause the whole operation to fail to occur at the top of the function, before we start calling `do_foo` type functions.  We don't have anything else in this function doing that yet, primarily because the validation is all being done via e.g. `REQ(validator=check_bool)` , but for this new validation, we should do it at the top of the function.